### PR TITLE
Remove references to "root" material

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR
PLA-7272

## Description 
RootInfo -> TerminalMaterialInfo
RootIdentifier -> TerminalMaterialIdentifier
preview_roots -> preview_materials

Companion nextgen-devkit PR: https://github.com/CitrineInformatics/nextgen-devkit/pull/890
### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/table_config.py [hunk 1]
```diff
@@ -404,21 +405,35 @@ def get_job_status(self, job_id: str):
...
 
         """
+        if preview_materials is None and preview_roots is None:
+            raise ValueError("Must specify preview materials")
+        if preview_materials is not None and preview_roots is not None:
+            raise ValueError("Cannot specify both preview_materials and preview_roots")
+        if preview_roots is not None:
+            preview_materials = preview_roots
+            warn("preview_roots argument is deprecated in favor of preview_materials",
+                 DeprecationWarning)
```
Comment (kroenlein) [inline src/citrine/resources/table_config.py:432]:
```suggestion
        if preview_roots is not None:
            warn("preview_roots argument is deprecated in favor of preview_materials",
                 DeprecationWarning)
            if preview_materials is None:
                preview_materials = preview_roots
            else:
                raise ValueError("Cannot specify both preview_materials and preview_roots")
        elif preview_materials is None:
                raise ValueError("Must specify preview materials")
```
I like that the above warns any time the user uses the `preview_roots` argument and, when it is removed, we delete a coherent block of text.

File: docs/source/data_extraction.rst [hunk 2]
```diff
@@ -227,8 +227,8 @@ There are several ways to define variables that take their values from Attribute
 
 * Identifiers
 
-  * :class:`~citrine.gemtables.variables.RootInfo`: for fields defined on the material at the terminal of the Material History, like the name of the material
-  * :class:`~citrine.gemtables.variables.RootIdentifier`: for the id of the Material History, which can be used as a unique identifier for the rows
+  * :class:`~citrine.gemtables.variables.TerminalMaterialInfo`: for fields defined on the material at the terminal of the Material History, like the name of the material
```
Comment (kroenlein) [inline docs/source/data_extraction.rst:230]:
```suggestion
  * :class:`~citrine.gemtables.variables.TerminalMaterialInfo`: for fields defined on the material at the terminus of the Material History, like the name of the material
```

**Review by kroenlein — APPROVED** [review]
Some fiddly stuff you can take or leave.  Why isn't it complaining about no version bump?

**Comment by bfolie** [comment]
> Why isn't it complaining about no version bump?

This is a merge into a working branch. We're collecting the breaking changes during this sprint, and then dropping them all at once.

This also means that I can accept your suggestion and then merge **without getting re-approval**

**Pull request merged**